File: backend/src/main/java/com/decmoe47/todo/service/TokenService.java
package com.decmoe47.todo.service;

import com.decmoe47.todo.model.vo.AuthenticationTokensVO;
import org.springframework.security.core.Authentication;

public interface TokenService {

    AuthenticationTokensVO generate(Authentication authentication);

    Authentication parse(String token);

    boolean isValidated(String token);

    AuthenticationTokensVO refresh(String refreshToken);

    void invalidate(String token);
}

File: backend/src/main/kotlin/com/decmoe47/todo/service/TokenService.kt
package com.decmoe47.todo.service

import com.decmoe47.todo.model.dto.SecurityUser
import com.decmoe47.todo.model.response.AuthenticationTokensResponse
import org.springframework.security.core.Authentication

interface TokenService {
    fun generate(authentication: Authentication): AuthenticationTokensResponse

    fun generate(user: SecurityUser): AuthenticationTokensResponse

    fun parse(token: String): Authentication

    fun isValid(token: String): Boolean

    fun refresh(refreshToken: String): AuthenticationTokensResponse

    fun invalidate(token: String)
}
